### PR TITLE
NSSC Solana updates

### PR DESCRIPTION
**Pull request by smoelius**
Several language fixes, as well as bug fixes to the owner and signer check examples.

### CODE REVIEW

File: not-so-smart-contracts/solana/improper_pda_validation/README.md [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
 
 View ToB's lint implementation for the bump seed canonicalization issue [here](https://github.com/crytic/solana-lints/tree/master/lints/bump_seed_canonicalization).
 
 ## Exploit Scenario
 
-In Solana, the `create_program_address` function creates a 32-byte address based off the set of seeds and program address. On it's own, the address may lie on the ed25519 curve. Consider the following without any other validation being referenced within a sensitive function, such as one that handles transfers. That PDA could be spoofed by a passed in user-controlled PDA with malicious functions to their own discretion.  
+In Solana, the `create_program_address` function creates a 32-byte address based off the set of seeds and program address. On its own, the point may lie on the ed25519 curve. Consider the following without any other validation being referenced within a sensitive function, such as one that handles transfers. That PDA could be spoofed by a passed in user-controlled PDA.
```
Comment (nop4e71) [inline not-so-smart-contracts/solana/improper_pda_validation/README.md:14]: "That PDA could be spoofed by a passed in user-controlled PDA."
This is what I couldn't put my finger on Friday. It isn't possible because there is no way to create and populate such an account.
1. If we try to create it as a "regular" account, the fact that it isn't a point on the curve means there is no private key, which means that no one can sign for it.
2. In fact, the only entity that can sign is the (targeted) program id, because `invoke_signed` uses the program id of the executing program as part of the seed.
In other words, to use a fake PDA with another bump, that PDS would have to have been created by the attacked program somehow.

File: not-so-smart-contracts/solana/ownership_check/README.md [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
 	let mut market = Market::unpack(&market_info.data.borrow())?;
- 
-	if &market.owner != current_owner.pubkey {
-    	    return Err(InvalidMarketOwner.into());
+
+	if &market.authority != current_authority.pubkey {
+    	    return Err(InvalidMarketAuthority.into());
 	}
-	market.owner = new_owner;
+	market.authority = new_authority;
```
Comment (nop4e71) [inline not-so-smart-contracts/solana/ownership_check/README.md:22]: I'm not sure about this.
If the "spoofed" account here is market, then we wouldn't need this instruction to set the auth of that (and in fact we couldn't because writing that property would make the tx fail unless ```market_info``` is owned by the contract)'.
If ```current_authority``` is the spoofed account, that would pretty much be impossible because an account with that id would already exists (or else that property value for ```market.authority``` wouldn't make sense.

**Review by nop4e71 — COMMENTED** [review]
I think we need to restructure the missing ownership.
Perhaps something like:
```
Pseudo code
instruction pay_escrow(accounts)
  if accounts.state.escrow_state == complete {
    tokens.transfer(escrow_vault, accounts.state.escrow_receiver, accounts.state.escrow_amount); 
  }
```
The working principle is some sort of escrow contract with a state and vault account.
Because of the missing ownership check on the passed state account, it is possible to make the contract believe the escrow is complete and transfer funds to the receiver.

File: not-so-smart-contracts/solana/ownership_check/README.md (hunk 2 above)
Comment (smoelius) [reply not-so-smart-contracts/solana/ownership_check/README.md]: > The working principle is some sort of escrow contract with a state and vault account. Because of the missing ownership check on the passed state account, it is possible to make the contract believe the escrow is complete and transfer funds to the receiver.

@nop4e71 I tried to realize your pseudocode. Nits welcome.

**Comment by smoelius** [comment]
From our perspective, this is ready for review, @montyly.
